x-pack/plugins/osquery/public/saved_queries/form/use_saved_query_form.tsx

```diff
@@ -41,17 +41,22 @@ interface UseSavedQueryFormProps {
   defaultValue?: SavedQuerySOFormData;
 }
 
-const deserializer = (payload: SavedQuerySOFormData): SavedQueryFormData => ({
-  id: payload.id,
-  description: payload.description,
-  query: payload.query,
-  interval: payload.interval ? parseInt(payload.interval, 10) : 3600,
-  snapshot: payload.snapshot,
-  removed: payload.removed,
-  platform: payload.platform,
-  version: payload.version ? [payload.version] : [],
-  ecs_mapping: !isEmpty(payload.ecs_mapping) ? payload.ecs_mapping : {},
-});
+const deserializer = (payload: SavedQuerySOFormData): SavedQueryFormData => {
+  // Way identify prebuilt Saved queries since they do not have nor snapshot nor removed (at the moment)
+  const hasNoResultTypes = payload.snapshot === undefined && payload.removed === undefined;
+
+  return {
+    id: payload.id,
+    description: payload.description,
+    query: payload.query,
+    interval: payload.interval ? parseInt(payload.interval, 10) : 3600,
+    snapshot: hasNoResultTypes ? true : payload.snapshot,
+    removed: payload.removed,
+    platform: payload.platform,
+    version: payload.version ? [payload.version] : [],
+    ecs_mapping: !isEmpty(payload.ecs_mapping) ? payload.ecs_mapping : {},
+  };
+};
```

Comment:
maybe?
```suggestion
const deserializer = (payload: SavedQuerySOFormData): SavedQueryFormData => ({
  id: payload.id,
  description: payload.description,
  query: payload.query,
  interval: payload.interval ? parseInt(payload.interval, 10) : 3600,
  snapshot: payload.snapshot ?? true,
  removed: payload.removed ?? false,
  platform: payload.platform,
  version: payload.version ? [payload.version] : [],
  ecs_mapping: !isEmpty(payload.ecs_mapping) ? payload.ecs_mapping : {},
});
```

Reply:
I think that could work, but I am still trying to think whether we need both values as undefined or just snapshot would be enough.

Reply:
Ok I think it's alright, thanks!